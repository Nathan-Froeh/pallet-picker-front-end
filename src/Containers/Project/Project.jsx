import React, { Component } from 'react';
import { fetchPalettesById, deleteProject } from '../../ApiCalls';
import { Palette } from '../../Components/Palette/Palette'

export class Project extends Component {
  constructor (props) {
    super(props)
    this.state = {
      project_name: this.props.name,
      project_id: this.props.id,
      palettes: []
    }
  }

  componentDidMount() {
    this.getPalettes()
  }

  handleDelete = (e) => {
    e.preventDefault()
    deleteProject(this.state.project_id)
  }

  getPalettes = () => {
    fetchPalettesById(this.state.project_id)
      .then(palettes => this.setState({palettes: palettes}))
      .catch(error => error.message)
  }

  render() {
    const addPalettes = this.state.palettes.map((palette, key) => {
      return <Palette 
      key={key}
      palette_id={palette.id}
      project_id={palette.project_id}
      name={palette.name}
      color_1={palette.color_1}
      color_2={palette.color_2}
      color_3={palette.color_3}
      color_4={palette.color_4}
      color_5={palette.color_5}/>
    })
    return (
      <div>
        <h3>{this.state.project_name}</h3>
        <button onClick={this.handleDelete}>Delete Project</button>
        {addPalettes}
      </div>
    )
  }
}

export default Project
