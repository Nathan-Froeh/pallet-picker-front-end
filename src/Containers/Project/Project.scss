.project {
  margin: 10px;
  background-color: rgba(169,169,169, 1);
  border: solid #2F4F4F 3px;
  -webkit-box-shadow: 6px 6px 5px 0px rgba(169,169,169, 1);
  -moz-box-shadow: 6px 6px 5px 0px rgba(169,169,169, 1);
  box-shadow: 6px 6px 5px 0px rgba(169,169,169, 1);
  border-radius: 10px;
}