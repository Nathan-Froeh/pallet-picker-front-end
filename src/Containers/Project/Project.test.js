import React from 'react';
import {shallow} from 'enzyme';
import {Project} from './Project';

describe('Project', () => {
  let wrapper
  const projects = [{name: 'Bat Cave'}]
  const palettes = [{
    name: 'floor',
    color_1: '#000000',
    color_2: '#000000',
    color_3: '#000000',
    color_4: '#000000',
    color_5: '#000000'
  }]
  const props = {
    projects: projects,
    palettes: palettes,
    deleteProject: jest.fn(),
    deletePalette: jest.fn(),
    getAllProjects: jest.fn()
  }
  beforeEach(() => {
    wrapper = shallow(<Project{...props}/>)
  })

  it('Project should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })


})