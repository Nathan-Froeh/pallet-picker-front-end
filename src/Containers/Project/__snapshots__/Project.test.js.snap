// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Project should match snapshot 1`] = `
<div
  className="project"
>
  <h3 />
  <button
    onClick={[Function]}
  >
    Delete Project
  </button>
  <Palette
    color_1="#000000"
    color_2="#000000"
    color_3="#000000"
    color_4="#000000"
    color_5="#000000"
    deletePalette={[MockFunction]}
    key="0"
    name="floor"
  />
</div>
`;
